logic/src/commonMain/kotlin/com/wire/kalium/logic/data/keypackage/KeyPackageLimitsProvider.kt

```diff
@@ -15,9 +18,16 @@ class KeyPackageLimitsProviderImpl(
     override val keyPackageUploadLimit: Int
         get() = if (kaliumConfigs.lowerKeyPackageLimits) KEY_PACKAGE_LIMIT_LOW else KEY_PACKAGE_LIMIT
 
-    override val keyPackageUploadThreshold: Float
+    private val keyPackageUploadThreshold: Float
         get() = KEY_PACKAGE_THRESHOLD
 
+    override fun needsRefill(keyPackageCount: Int): Boolean {
+        return keyPackageCount < (keyPackageUploadLimit * keyPackageUploadThreshold)
+    }
+
+    override fun refillAmount(currentAmount: Int) =
+        keyPackageUploadLimit - currentAmount
```

Comment:
The use case changed so we should always refill the whole `keyPackageUploadLimit` now. See https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/556499553/Use+case+refill+key+packages+MLS#2.-Upload-new-key-packages for details.

Reply:
Thanks, just kept the function as function to make it easy to modify/add logic later.